Example/Tests/ContentMetadataTest.swift

```diff
@@ -114,22 +114,36 @@ class ContentMetadataTest: QuickSpec {
 
             context("when updating session") {
                 it("update video duration") {
+                    _ = TestDouble(aClass: playerDouble!, name: "duration", return: 50.0)
+
                     playerDouble.fakePlayEvent() // to initialize session
-                    var metadata = MetadataOverrides()
-                    metadata.duration = 50
-                    convivaAnalytics.updateContentMetadata(metadataOverrides: metadata)
                     let spy = Spy(aClass: CISVideoAnalyticsTestDouble.self, functionName: "setContentInfo")
 
                     expect(spy).to(
                         haveBeenCalled(withArgs: ["duration": "50"])
                     )
                 }
 
+                describe("when duration is not yet available on play") {
+                    it("update video duration on source loaded") {
+                        _ = TestDouble(aClass: playerDouble!, name: "duration", return: 0.0)
+                        playerDouble.fakePlayEvent() // to initialize session
+
+                        _ = TestDouble(aClass: playerDouble!, name: "duration", return: 50.0)
+                        playerDouble.fakeSourceLoadedEvent() // to initialize session
+
+                        let spy = Spy(aClass: CISVideoAnalyticsTestDouble.self, functionName: "setContentInfo")
+
+                        expect(spy).to(
+                            haveBeenCalled(withArgs: ["duration": "50"])
+                        )
+                    }
+                }
+
                 it("update stream type (VOD/Live)") {
+                    _ = TestDouble(aClass: playerDouble!, name: "isLive", return: true)
+
                     playerDouble.fakePlayEvent() // to initialize session
-                    var metadata = MetadataOverrides()
-                    metadata.streamType = StreamType.CONVIVA_STREAM_LIVE
-                    convivaAnalytics.updateContentMetadata(metadataOverrides: metadata)
                     let spy = Spy(aClass: CISVideoAnalyticsTestDouble.self, functionName: "setContentInfo")
```

Comment:
This test was testing the `MetadataOverrides` feature instead of the default metadata collection feature.